pycbc/inference/evidence.py

```diff
@@ -0,0 +1,465 @@
+                     The inverse temperatures used in the MCMC.
+
+    Returns
+    ----------
+    log_evidence    : float
+                    Estimation of the log of the evidence.
+    mcmc_std: float
+                    The standard deviation of the evidence estimate
+                    based on the deviation of log likelihoods for
+                    each temperature.
+    """
+    # Reverse order sort
+    order = numpy.argsort(betas)[::-1]
+    betas = betas[order]
+    log_likelihood = log_likelihood[order]
+
+    # Assume log likelihood is given in shape of beta,
+    # walker, iteration.
+    log_likelihood = numpy.reshape(log_likelihood, (len(betas),
+                                                    len(log_likelihood[0].flatten())))
+
+    log_rk_pb = numpy.zeros(len(betas) - 1)
+    for i in range(len(betas) - 1):
+        betas_diff = betas[i] - betas[i+1]
+        # Max log likelihood for beta [i+1]
+        max_logl_pb = numpy.max(log_likelihood[i+1])
+        val_1 = betas_diff * max_logl_pb
+        val_2 = numpy.log(numpy.average(numpy.exp(betas_diff * (log_likelihood[i+1] - max_logl_pb))))
+
+        log_rk_pb[i] = val_1 + val_2
+    log_rk = numpy.sum(log_rk_pb)
+    log_evidence = log_rk
+
+    mcmc_std = 0
+    for i in range(len(betas) - 1):
+        betas_diff = betas[i] - betas[i+1]
+
+        pre_fact = numpy.exp((betas_diff * log_likelihood[i+1]) - log_rk_pb[i]) - 1.0
+        val_2 = numpy.sum(pre_fact ** 2)
+
+        mcmc_std += val_2
+
+    mcmc_std /= float(len(log_likelihood[0])) ** 2.0
+    mcmc_std = numpy.sqrt(mcmc_std)
+
+    return log_evidence, mcmc_std
+
+def calculate_log_bayes_factor(log_ev_1, log_ev_2):
+    """
+    Calculate the point-wise log Bayes factor from the difference
+    in the logarithm of the model evidences.
+
+    Parameters
+    ----------
+    log_ev_1 : float
+               The logarithm of the evidence for model 1
+
+    log_ev_2 : float
+               The logarithm of the evidence for model 2
+
+    Returns
+    ----------
+    log_bayes_factor : float
+               The logarithm of the Bayes factor for model 1 vs
+               model 2.
+    """
+    log_bayes_factor = log_ev_1 - log_ev_2
+    return log_bayes_factor
+
+def calculate_bayes_factor(log_ev_1, log_ev_2):
+    """
+    Calculate the point-wise Bayes factor from the exponential
+    of the difference in the logarithm of the model evidences.
+    Mathematically equivalent to the ratio of the model
+    evidences.
+
+    Parameters
+    ----------
+    log_ev_1 : float
+               The logarithm of the evidence for model 1.
+
+    log_ev_2 : float
+               The logarithm of the evidence for model 2.
+
+    Returns
+    ----------
+    bayes_factor : float
+               The Bayes factor for model 1 vs model 2.
+    """
+    return numpy.exp(calculate_log_bayes_factor(log_ev_1,
+                                                log_ev_2))
+
+def sim_log_bayes_factor_distr(log_ev_1, std_log_ev_1,
```

Comment:
It might makes sense to put these functions in a separate module, as they are really test functions.

Reply:
If they are really test functions, would it make sense to put them into a unit test?

Reply:
I've removed these functions. Not sure what's meant by them being test functions. They're just conversions from log B to B.